docs/sdg/knowledge-doc-ingestion.md

```diff
@@ -0,0 +1,96 @@
+# Knowledge Document Ingestion Pipeline Design Proposal
+
+**Author**: Aakanksha Duggal
+
+## 1. Introduction
+
+As part of extending InstructLab's capabilities, this pipeline is designed to support ingestion and processing of various document formats such as Markdown, PDF, DOCX, and more. The goal is to create a unified, modular system that seamlessly integrates with **Synthetic Data Generation (SDG)** and **Retrieval-Augmented Generation (RAG)** workflows, simplifying the process for users while maintaining flexibility for future enhancements.
+
+## 2. Use Case
+
+To enable the ingestion and processing of a wide range of document types, this pipeline must handle formats including:
+
+- Markdown (MD)
+- PDF
+- TXT
+- ASCII Docs
+- DOCX
+- HTML
+- PPTX
+
+This proposal outlines how to build a pluggable system that accommodates these formats and integrates effectively into existing **SDG** and **RAG** workflows.
+
+## 3. Proposed Approach
+
+### 3.1 Custom InstructLab Schema Design
+
+- **Objective**: Define a custom "instructlab" schema that standardizes input formats for SDG and RAG pipelines. This schema will bridge the gap between various document types and the specific formats required for further processing.
+- **Modularity**: The system should allow easy extension, enabling support for new document types and processing workflows without disrupting core functionality.
+  
+The instructlab schema will serve as the intermediary format, supporting flexibility while ensuring compatibility with the ingestion process.
+
+### 3.2 PDF and Document Conversion via Docling
+
+- **Docling Integration**: We will leverage **Docling** to convert files into structured JSON, which will be the starting point for the instructlab schema. Individual components will post-process this JSON as per the requirements of the specific SDG and RAG workflows.
+  
+- **Docling v2**: We have collaborated with the Docling team to extend their tool’s capabilities, allowing conversion for PDF, HTML and DOCX formats. These new document types will be supported in the ingestion pipeline via docling's upgraded v2 release.
+
+### 3.3 Introducing the Document Chunking Command
+
+- **Command Overview**: We propose a new command, `ilab document format`, which will:
```

Comment:
Is there a particular reason for using the word `format`? The first time I read the command without reading the whole document, it gave me the impression the command was for transforming a document from one format to another format (e.g. pdf to md, or pdf to json). Should we consider a word or verb that more closely resembles what is actually happening in this step? For example:

`ilab docs import --input path/to/document.pdf --output path/to/schema`
`ilab docs ingest --input path/to/document.pdf --output path/to/schema`
`ilab docs process --input path/to/document.pdf --output path/to/schema`

Reply:
I propose `ilab data` being used instead of `ilab docs` or `ilab document` - we already have this command group implemented in the CLI, and it would be good if users can keep their data manipulation in a single command group. I'd like some folks from UX to weigh-in here as well.

Reply:
@williamcaban I agree with your sentiments about `format` its very ambiguous. I think you are spot on with your suggestions. Here are some suggestions based on each prompt to provide some clarity for the user. 

`ingest` Provide the document path to ingest and process into the desired scheme.
`process` Provide the document path to process into the scheme.
`import`Specifiy the document path to import and format according to the scheme requirements.
`chunk` Enter the document path to split and structure for the scheme.

Reply:
If we think the document-related commands are going to expand considerably, then this might be the time to create a new command group. If the document processing will remain a small subset, then integrating them under the data group would simplify the CLI structure. 

`ingest` Provide the document path to ingest and process into the desired scheme.
`process` Provide the document path to process into the scheme.
`import` Specify the document path to import and format according to the scheme requirements.
`chunk` Enter the document path to split and structure for the scheme.

Reply:
If think the **document-related commands** are going to expand considerably, then this might be the time to create a new command group. If the document processing will remain **a small subset**, then **integrating** them under the `data` group would simplify the CLI structure. 

`ingest` Provide the document path to ingest and process into the desired scheme.
`process` Provide the document path to process into the scheme.
`import` Specify the document path to import and format according to the scheme requirements.
`chunk` Enter the document path to split and structure for the scheme.